tools/codemod/src/utils/Extensions.ts

```diff
@@ -2,33 +2,134 @@
+  path(): ASTPath<N>;
+  node(): N;
+  first(): this;
+  nonEmpty(): this | undefined;
+  single(): this | undefined;
+  navigatePath(...path: (string | number)[]): Collection<any>;
+  findInChildren<T extends ASTNodeType>(
+    type: Type<T>,
+    filter?: RecursiveMatchNode<T>,
+    pathName?: string
+  ): Collection<T>;
+  forEachAsCollection(
+    callback: (path: Collection<N>, i: number, paths: ASTPath<N>[]) => void
+  ): this;
+  filterPath(pathName: string): this;
+  filterType<T extends ASTNodeType>(type: Type<T>): Collection<T>;
+  removeFromParentPath(): this;
   contains(other: Collection): boolean;
   concat(other: Collection): this;
   rename(source: string, target: string): this;
 }
 
 function extensionsPlugin(j: JSCodeshift) {
-  const globalMethods: GlobalMethods = {
+  const globalMethods: GlobalMethods<ASTNode> = {
+    /** Returns the first path in the collection */
+    path(this: Collection) {
+      return this.paths()[0];
+    },
+    /** Returns the first node in this collection */
+    node(this: Collection) {
+      return this.nodes()[0];
+    },
+    first(this: Collection) {
+      return this.at(0);
+    },
+    nonEmpty(this: Collection) {
+      return this.length > 0 ? this : undefined;
+    },
+    single(this: Collection) {
+      return this.length === 1 ? this : undefined;
+    },
+    navigatePath(this: Collection, ...path) {
+      return j(this.get(...path));
+    },
+    /** An alternative to find that only traverses paths of the immediate children of each path in the collection */
+    findInChildren(this: Collection, type, filter, pathName) {
+      const paths: ASTPath[] = [];
+      const visitorMethodName = "visit" + type;
+
+      this.forEach((parentPath: ASTPath) => {
+        const visitor = {
+          [visitorMethodName]: function (path: ASTPath) {
+            if (path === parentPath) {
+              this.traverse(path);
+              return false;
+            }
+
+            if (!path.parentPath || path.parentPath.node !== parentPath.node)
+              return false;
+
+            if (pathName && !isFromPath(path, pathName)) return false;
+
+            if (!filter || matchNode(path.value, filter)) paths.push(path);
+
+            return false;
+          },
+        };
+        j.types.PathVisitor.visit(parentPath as any, visitor);
+      });
+      return j(paths);
+
+      // Implementation with jscodeshift api only
+      /* return this.map((parentPath) =>
+        j(parentPath)
+          .find(type, filter)
+          .filter((childPath) => {
+            if (childPath.parentPath.node !== parentPath.node) return false;
+            if (nodePathName) {
+              const childNodePathName =
+                typeof childPath.name === "number"
+                  ? childPath.parentPath.name
+                  : childPath.name;
+              return nodePathName === childNodePathName;
+            }
+            return true;
+          })
+          .paths()
+      );*/
```

Comment:
Is it better ? Is it a reference ? Should this be used or the comment removed ?

Reply:
The commented out code is theoretically slower since it finds all children of children of children... and then filters only the first child layer. The code above is a tweaked implementation of jscodeshift find method so it would stop after finding only the first layer of children (immediate children). So it is up to you to decide which one to keep (above one is faster, commented one is more readable)